backend/handlers/math/probabilities/wpam/wpam_marketprobabilities.go

```diff
@@ -51,3 +54,14 @@ func CalculateMarketProbabilitiesWPAM(marketCreatedAtTime time.Time, bets []mode
 
 	return probabilityChanges
 }
+
+func ProjectNewProbabilityWPAM(marketCreatedAtTime time.Time, currentBets []models.Bet, newBet models.Bet) ProjectedProbability {
+
+	updatedBets := append(currentBets, newBet)
+
+	probabilityChanges := CalculateMarketProbabilitiesWPAM(marketCreatedAtTime, updatedBets)
```

Comment:
🤔 I'm wondering if we need to use `CalculateMarketProbabilitiesWPAM` this way, we just end up throwing away most of the results on line 64. I think we could rework this and just get the bet totals before passing them to this function: https://github.com/openpredictionmarkets/socialpredict/pull/284/files#diff-218fc849d1491bbf985665580df93e8ec74952e28c33653cbaafaa4c5f19c5aeR44-R49

Reply:
Either way, the previous function still used the, "final," probability, though it was calculated upon request earlier when the individual viewing the market in question got a list of all historical probabilities, which invoked wpam on the backend.

The difference is, wpam was only invoked one time, on the front end, it was, "cached," in React.

However, the disadvantage of that is that if someone was trading on the market simultaneously, the result would be less accurate, because if someone made a trade, the result on the react page would be calculated incorrectly.

This method, which uses wpam directly, recalculates every time so if someone makes a bet in the interim, the result would show correctly when someone calls the API (or hits the button on the frontend).